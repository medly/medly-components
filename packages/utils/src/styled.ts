import * as styledComponents from 'styled-components';

const { default: styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme } = styledComponents;

export { styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme };
