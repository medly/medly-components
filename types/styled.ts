import 'styled-components';
import { Theme } from '../packages/theme';

declare module 'styled-components' {
    export interface DefaultTheme extends Theme {}
}
