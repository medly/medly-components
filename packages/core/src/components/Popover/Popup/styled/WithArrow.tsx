import { css } from 'styled-components';
import { PopoverPopupProps } from '../types';

const getPositionWithArrow = ({ placement }: PopoverPopupProps) => {
    switch (placement) {
        case 'top':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%);
                    top: 100%;
                    left: 50%;
                }
            `;
        case 'top-start':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    left: 2rem;
                }
            `;
        case 'top-end':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    right: 2rem;
                }
            `;
        case 'right':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(90deg);
                    top: 50%;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-start':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    top: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-end':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    bottom: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'bottom':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%) rotate(180deg);
                    left: 50%;
                    bottom: 100%;
                }
            `;
        case 'bottom-start':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    left: 2rem;
                    bottom: 100%;
                }
            `;
        case 'bottom-end':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    right: 2rem;
                    bottom: 100%;
                }
            `;
        case 'left':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(270deg);
                    top: 50%;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-start':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    top: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-end':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    bottom: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
    }
};

export const getStyleWithArrow = css<PopoverPopupProps>`
    border-radius: 0.8rem;
    padding: 1.5rem 2rem;

    &::after {
        content: '';
        width: 0;
        height: 0;
        z-index: 4;
        position: absolute;
        border-left: 0.8rem solid transparent;
        border-right: 0.8rem solid transparent;
        border-top: 1rem solid ${({ bgColor }) => bgColor};
    }

    ${getPositionWithArrow}
`;
