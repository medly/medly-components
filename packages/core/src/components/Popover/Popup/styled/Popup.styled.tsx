import { fullHeight, fullWidth } from '@medly-components/utils';
import styled, { css } from 'styled-components';
import { Placement } from '../../types';
import { StyledPopoverPopupProps } from '../types';
import { getStyleWithArrow } from './WithArrow';

export const getPosition = ({ placement, distance = '0px' }: { placement: Placement; distance?: string }) => {
    switch (placement) {
        case 'top-start':
            return css`
                bottom: calc(100% + ${distance});
                left: 0%;
            `;
        case 'top':
            return css`
                bottom: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'top-end':
            return css`
                bottom: calc(100% + ${distance});
                right: 0%;
            `;
        case 'right-start':
            return css`
                top: 0%;
                left: calc(100% + ${distance});
            `;
        case 'right':
            return css`
                left: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'right-end':
            return css`
                bottom: 0;
                left: calc(100% + ${distance});
            `;
        case 'bottom-start':
            return css`
                top: calc(100% + ${distance});
                left: 0%;
            `;
        case 'bottom':
            return css`
                top: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'bottom-end':
            return css`
                top: calc(100% + ${distance});
                right: 0%;
            `;
        case 'left-start':
            return css`
                right: calc(100% + ${distance});
                top: 0%;
            `;
        case 'left':
            return css`
                right: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'left-end':
            return css`
                right: calc(100% + ${distance});
                bottom: 0%;
            `;
    }
};

export const PopupStyled = styled('div')<StyledPopoverPopupProps>`
    z-index: 4;
    position: absolute;
    background-color: ${({ bgColor }) => bgColor};

    ${props => props.fullWidth && fullWidth()};
    ${props => props.fullHeight && fullHeight()};

    ${props => getPosition({ placement: props.placement, distance: props.withArrow ? '1.5rem' : props.distance })}
    ${({ withArrow }) => withArrow && getStyleWithArrow}
`;
