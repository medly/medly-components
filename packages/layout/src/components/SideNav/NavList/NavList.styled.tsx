import { defaultTheme } from '@medly-components/theme';
import styled from 'styled-components';

export const NavList = styled('ul')`
    flex: 1;
    padding: 0;
    margin: 0;
    list-style-type: none;
    overflow-x: hidden;
    overflow-y: auto;

    &::-webkit-scrollbar {
        width: 0.5rem;
        background-color: ${({ theme }) => theme.sideNav.bgColor};
    }

    &::-webkit-scrollbar-thumb {
        border-radius: 1rem;
        background-color: ${({ theme }) => theme.sideNav.scrollBarColor};
    }
`;

NavList.displayName = 'NavList';
NavList.defaultProps = {
    theme: defaultTheme
};
