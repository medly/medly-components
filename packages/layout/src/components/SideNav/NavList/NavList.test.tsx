import { cleanup, render } from '@test-utils';
import { NavList } from './NavList.styled';

describe('NavList', () => {
    afterEach(cleanup);

    it('should render properly', () => {
        const { container } = render(<NavList>Dummy</NavList>);
        expect(container).toMatchSnapshot();
    });
});
