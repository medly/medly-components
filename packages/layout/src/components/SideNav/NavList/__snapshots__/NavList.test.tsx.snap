// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList should render properly 1`] = `
.c0 {
  flex: 1;
  padding: 0;
  margin: 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
}

.c0::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c0::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

<div>
  <ul
    class="c0"
  >
    Dummy
  </ul>
</div>
`;
